### ISSUE TITLE
[Question] still receive song when ptt is pressed ?

### ISSUE THREAD

**Issue by spilz87** [dchote/talkiepi#20]
Hello,

I'm looking for something similar or maybe exactly this project but I want to be sure before buying more raspberry pis and spending time on it :)

Simple question :

When someone press the button on his side, does he still receive the voice from other side ? Or pressing button turns off speaker (like on analogic Talki-walki)?

(My need is to still have speaker working)

Thanks in advance for your reply 

Regards

Spilz

**Comment by spilz87** [comment]
No one is able to reply to this question ?

**Comment by zrdimetc** [comment]
@spilz87, why dont you try with your Raspberry Pi talkiepi build and some other Mumble client, like [Plumble](https://github.com/acomminos/Plumble), [Mumla](https://gitlab.com/quite/mumla), [official Mumble client](https://www.mumble.info/downloads/)? Is the audio working for your intended logic? Run talkiepi against any other Mumble client. Can you talk and listen at the same time from both clients, talkiepi and any other Mumble client? You would like to have a full "duplex" communication between the devices? When the users can talk and listen at the same time, like in a telephone conversation? Unlike the radio communication, which is "semi-duplex" or "half duplex" or sometimes called "simplex", when one user is talking, then listening for the other user, then talking again, alternatively? talkiepi does not mute audio output during a PTT press to emulate radio-like half-duplex communication. So yes, you should be able to transmit over the other client and the audio output (speaker) in talkiepi won't be muted. There are talkiepi forks like [talkkonnect](https://github.com/talkkonnect) which support choosing between the "duplex" phone-like communication and "half-duplex" - radio-like communication. Unless you use a headset in duplex communication, with external speaker and microphone, depending on audio gain and proximity, you may experience "echo". In some use cases the audio muting is preferable and there will be cases where the duplex communication without muting is better. [talkkonnect](https://talkkonnect.com) supports both. Official Mumble client can also manipulate volume and attenuate other applications and users when you are talking.

**Comment by spilz87** [comment]
Thanks for your reply 
According to your description I'm looking for full duplex system. My project is to connect 2 old phones through wifi.

So pick up the phone will do like pressing talk button, that's why I wanted to be sure if both phones are picked up it still work 🤷‍♂️

So before trying to build a talkypi I wanted to be sure it does want I need. 

Thanks

**Comment by zrdimetc** [comment]
Yes, you can have a very nice 2-way IP intercom system, after a successful talkiepi build. The intercom press button logic is  different from the phone on-hook off-hook logic. If you want to convert phone silence hook switch to work as a PTT button, check which switch pins are closed on [handset pickup](http://lucidscience.com/pro-phone%20voice%20changer-5.aspx)? Hook switch is usually DPDT (Double Pole Double Throw). You may have to alter talkiepi PTT button logic in the code. Or use some relay. transistor, IC to alter the phone hook switch logic? Or just try rewire the switch for working. How will you signal to the other client that there is an incoming call? If the phone handset is not picked, will you be able to hear the audio clearly from the other side? talkiepi doesnt have a LED to indicate active incoming stream. RX led can be engineered with some timer function inside `stream.go`. [talkkonnect](https://github.com/talkkonnect) has a fully working RX LED logic. If the 2 devices are on your local network, you could run audio streams between them continuously without a need for pressing the PTT, but how to signal to the other side to pick the handset? You could try to emulate "phone ringing" by creating some audible or visible alerting function? Or do you plan to use a speaker in the phone for audio output and microphone in handset for input? Maybe route an audio from phone speaker to handset after the handset pickup with some switching? Good luck with your build. Best regards.
